model-datastructure/src/commonMain/kotlin/org/modelix/model/operations/OTBranch.kt

```diff
@@ -30,30 +30,47 @@ class OTBranch(
     private val idGenerator: IIdGenerator,
     private val store: IDeserializingKeyValueStore,
 ) : IBranch {
-    private var operations: MutableList<IAppliedOperation> = ArrayList()
-    private var treeForOperations: ITree = branch.computeRead { branch.transaction.tree }
-    private val operationsLock = Any()
+    private var currentOperations: MutableList<IAppliedOperation> = ArrayList()
+    private val completedChanges: MutableList<Pair<List<IAppliedOperation>, ITree>> = ArrayList()
     private val id: String = branch.getId()
 
     fun operationApplied(op: IAppliedOperation) {
-        runSynchronized(operationsLock) {
-            operations.add(op)
-            treeForOperations = transaction.tree
-        }
+        check(canWrite()) { "Only allowed inside a write transaction" }
+        currentOperations.add(op)
     }
 
     override fun getId(): String {
         return id
     }
 
+    @Deprecated("renamed to getPendingChanges()", ReplaceWith("getPendingChanges()"))
     val operationsAndTree: Pair<List<IAppliedOperation>, ITree>
         get() {
-            runSynchronized(operationsLock) {
-                val newOperations: List<IAppliedOperation> = operations
-                operations = ArrayList()
-                return Pair(newOperations, treeForOperations)
+            return runSynchronized(completedChanges) {
+                val result = when (completedChanges.size) {
+                    0 -> emptyList<IAppliedOperation>() to computeReadT { it.tree }
+                    1 -> completedChanges[0]
+                    else -> completedChanges.flatMap { it.first } to completedChanges.last().second
+                }
+                completedChanges.clear()
+                result
```

Comment:
This is the same implementation as in `getPendingChanges`. Can't this getter redirect to that method to avoid the duplication?